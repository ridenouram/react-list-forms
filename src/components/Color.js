import React from 'react';
const hexRgb = require('hex-rgb');
import PropTypes from 'prop-types';

export default function Color({ name, hex }) {
  Color.propTypes = {
    name: PropTypes.string.isRequired,
    hex: PropTypes.string.isRequired
  };

  const rgb = hexRgb(hex);

  return (
    <dl>
      <div style={{ background: hex, height: '40px', width: '40px' }}></div>
      <dt>Name</dt>
      <dd>{name}</dd>

      <dt>Hex</dt>
      <dd>{hex}</dd>

      <dt>RGB</dt>
      <dd>red: {rgb.red}</dd>
      <dd>green: {rgb.green}</dd>
      <dd>blue: {rgb.blue}</dd>
    </dl>
  );
}
