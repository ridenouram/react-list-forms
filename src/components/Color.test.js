import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Color name="blue" hex="#60b2e5" rgb={{ r: 96, g: 178, b: 229 }} />);
    expect(wrapper).toMatchSnapshot();
  });
});

