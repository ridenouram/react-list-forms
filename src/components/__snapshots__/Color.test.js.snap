// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component matches snapshot 1`] = `
<dl>
  <div
    style={
      Object {
        "background": "#FF0000",
        "height": "40px",
        "width": "40px",
      }
    }
  />
  <dt>
    Name
  </dt>
  <dd>
    Red
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #FF0000
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    red: 
    255
  </dd>
  <dd>
    green: 
    0
  </dd>
  <dd>
    blue: 
    0
  </dd>
</dl>
`;
